MAINT: Add benchmarking files (#55)
nipreps/nifreeze: benchmarks/README.rst

..  -*- rst -*-

===================
NiFreeze benchmarks
===================

> 

Reviewer: Rst does not leave empty lines after titles:
```suggestion
```

Benchmarking NiFreeze with Airspeed Velocity.

Usage
-----

> 

Reviewer: ```suggestion
```

Airspeed Velocity manages building and Python environments by itself,

> unless told otherwise. To run the benchmarks, you do not need to
> install a development version of NiFreeze to your current Python environment.

Reviewer: Semantic line breaks:
```suggestion
unless told otherwise.
To run the benchmarks, you do not need to install
a development version of *NiFreeze* on your current
*Python* environment.
```

Author: We should add an rst linter: different people will disagree on this.

Author: > A linter cannot deal with semantics...

OK, valid point, but my point is to reduce reviewing and maintenance burden, and avoiding inconsistencies across all NiPreps. It is even hard to enforce this within a single repository. I'd rather adopt a linter and content ourselves with however that linter breaks lines, even if they are not semantic breaks. Requires effort to get used to probably, but involves less maintenance.

https://pypi.org/project/restructuredtext-lint/
https://pypi.org/project/sphinx-lint/

Maybe there are better ones as well.

> The question is whether you or other maintainer disagrees :)

:white_flag: if I am the only one disagreeing? Topic for tech call?

> To run all benchmarks for the latest commit, navigate to the root NiFreeze

Reviewer: ```suggestion
To run all benchmarks for the latest commit, navigate to *NiFreeze*'s root
```

``benchmarks`` directory and execute::

    asv run

For testing benchmarks locally, it may be better to run these without
replications::

    export REGEXP="bench.*Ufunc"
    asv run --dry-run --show-stderr --python=same --quick -b $REGEXP

All of the commands above display the results in plain text in the console,

> and the results are not saved for comparison with future commits. For
> greater control, a graphical view, and to have results saved for future
> comparison you can run ASV commands (record results and generate HTML):::

Reviewer: ```suggestion
and the results are not saved for comparison with future commits.
For greater control, a graphical view, and to have results saved for future
comparisons, you can run ASV as follows to record results and generate
the HTML reports::
```

    asv run --skip-existing-commits --steps 10 ALL
    asv publish
    asv preview

> More on how to use ``asv`` can be found in `ASV documentation`_

Reviewer: ```suggestion
More on how to use ``asv`` can be found in the `ASV documentation`_.
```
